### PR TITLE
devel/valgrind: Add at v3.20.0

### PR DESCRIPTION
**Pull request by der-eismann**
Tested by enabling checks for libseccomp
```
=============== Wed Jan 11 06:14:22 PM UTC 2023 ===============
Regression Test Report ("regression ")
 batch name: 01-sim-allow
 test mode:  c
 test type:  bpf-sim
Test 01-sim-allow%%001-00001 result:   SUCCESS
Test 01-sim-allow%%001-00002 result:   SUCCESS
Test 01-sim-allow%%001-00003 result:   SUCCESS
...
Test 59-basic-empty_binary_tree%%001-00350 result:   SUCCESS
Test 59-basic-empty_binary_tree%%001-00351 result:   SUCCESS
 test mode:  c
 test type:  bpf-valgrind
Test 59-basic-empty_binary_tree%%002-00001 result:   SUCCESS
Regression Test Summary
 tests run: 8609
 tests skipped: 46
 tests passed: 8609
 tests failed: 0
 tests errored: 0
============================================================
PASS: regression
=============
1 test passed
=============
```

### CODE REVIEW

File: devel/valgrind/stone.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#
+# SPDX-FileCopyrightText: © 2020-2023 Serpent OS Developers
+#
+# SPDX-License-Identifier: Zlib
+#
+name        : valgrind
```
Comment (ReillyBrogan) [inline devel/valgrind/stone.yml:6]: I messed around with enabling checks. It looks like it needs docbook-sgml and/or docbook-xsl to run:

```
name        : valgrind
version     : '3.20.0'
release     : 1
homepage    : https://valgrind.org/
upstreams   :
    - https://sourceware.org/pub/valgrind/valgrind-3.20.0.tar.bz2 : 8536c031dbe078d342f121fa881a9ecd205cb5a78e639005ad570011bdb9f3c6
summary     : Instrumentation framework for testing & debugging
description : |
    Valgrind is an instrumentation framework for building dynamic analysis tools.
    There are Valgrind tools that can automatically detect many memory management
    and threading bugs, and profile your programs in detail.
license     :
    - GPL-2.0-or-later
builddeps   :
    - gcc-devel
checkdeps   :
    - binary(ps)
    - glibc-dbginfo
rundeps     :
    - glibc-dbginfo
setup       : |
    %configure CFLAGS="$CFLAGS -fno-stack-protector"
build       : |
    %make
install     : |
    %make_install
check       : |
    # Make sure a basic binary runs. There should be no errors.
    ./vg-in-place --error-exitcode=1 /usr/bin/true

    # Make sure no extra FLAGS leak through, the testsuite
    # sets all flags necessary. See also configure above.
    ## TODO this needs docbook-sgml and docbook-xsl to run
    # %make check CPPFLAGS= CFLAGS= CXXFLAGS= LDFLAGS=
```

File: libs/libseccomp/stone.yml [hunk 2]
```diff
@@ -15,14 +15,14 @@ description : |
...
 setup       : |
     %configure --disable-static
 build       : |
     %make
 install     : |
     %make_install
-#check       : |
-# TODO enable checks after packaging valgrind
-#    %make check
+check       : |
```
Comment (ReillyBrogan) [inline libs/libseccomp/stone.yml:27]: I think this is worth a relbump for libseccomp.

File: libs/libseccomp/stone.yml [hunk 3]
```diff
@@ -15,14 +15,14 @@ description : |
 license     : LGPL-2.1-only
 builddeps   :
     - binary(gperf)
-#checkdeps   :
-#    - valgrind
+checkdeps   :
+    - binary(which) # make check uses 'which' to test if 'head' is available
+    - pkgconfig(valgrind)
```
Comment (ReillyBrogan) [inline libs/libseccomp/stone.yml:20]: Actually just needs `binary(valgrind)`.

**Review by ReillyBrogan — CHANGES_REQUESTED** [review]

**Comment by ikeycode** [comment]
So I think we need an option to disable hardening in our tuning ?

**Comment by ikeycode** [comment]
I'm hopeful this behaviour will help when we land in volatile

https://github.com/serpent-os/boulder/commit/84aed1269122eb3f6212bb6e1a9cff8de9091c87

**Comment by ReillyBrogan** [comment]
> I'm hopeful this behaviour will help when we land in volatile
> 
> [serpent-os/boulder@84aed12](https://github.com/serpent-os/boulder/commit/84aed1269122eb3f6212bb6e1a9cff8de9091c87)

Yeah, @der-eismann  try this with the following additional option now:
```
tune:
  - harden-none
```

**Comment by ikeycode** [comment]
commandeering

**Comment by ikeycode** [comment]
all included now - tyvm! :D

**Pull request merged**